src/modules/dashboard/components/CoinMachine/CoinMachine.tsx

```diff
@@ -170,17 +170,11 @@ const CoinMachine = ({
     if (!saleTokensData || !periodTokensData || !hasSaleStarted) {
       return undefined;
     }
-    const maxPerPeriodTokens = bigNumberify(maxPerPeriod);
-    const leftAvailableTokens = bigNumberify(
-      coinMachineTokenBalanceData?.coinMachineTokenBalance || '0',
-    );
 
     return {
       decimals: saleTokensData.coinMachineSaleTokens.sellableToken.decimals,
       soldPeriodTokens: bigNumberify(activeSold),
-      maxPeriodTokens: maxPerPeriodTokens.gt(leftAvailableTokens)
-        ? leftAvailableTokens
-        : bigNumberify(maxPerPeriod),
```

Comment:
If you remove this comparison, then if in coin machine there are less tokens than maxPerPeriod, we will still show maxPerPeriod while we should be showing what is actually left in coin machine. Does it make sense? People can't buy more than what we have in coin machine even if maxPerPeriod shows more.

Reply:
Ooof my bad, I didn't think about that case. 

I will have to tweak this logic and/or tweak the sold-out logic because this is making the coin machine assume that the tokens are sold out if the `maxPeriodTokens` is less or equal to the sold amount of tokens (Right now we are **always** assuming that `maxPeriodTokens` is the _full_ amount, without considering the possibility that it could also be what's left.) 

Tricky.